src/trading/simulation/simulate-buy.js

```diff
@@ -32,6 +34,10 @@ function simulateBuy(outcome, sharesToCover, shareBalances, tokenBalance, userAd
     }
   }
 
+
+  // Worst-case settlement fee for the buyer is if they buy at minPrice
+  simulatedBuy.maxSettlementFees = calculateSettlementFee(sharesToCover, marketCreatorFeeRate, maxPrice.minus(minPrice), shouldCollectReportingFees, reportingFeeRate, minPrice);
```

Comment:
The price at which the buyer is buying should be set by the prices of the orders they're buying, not `minPrice`.  `minPrice` is the minimum possible price allowed on this market.